### PR TITLE
Add Repolinter Check Workflow to All Tiers

### PR DESCRIPTION
**Pull request by IsaacMilarky**
## Problem

Currently, new repositories generated through repo-scaffolder will not have any workflows associated with them to start. 

## Solution

Now, new repositories built through repo-scaffolder that are above tier 0 will start with a checks.yml workflow file that will check the repolinter rules that we have defined and file issues on the repository accordingly. 

## Result

There are new workflow files for each tier except for tier 0 as it doesn't have a `repolinter.json` file associated with it.

### CODE REVIEW

File: tier1/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          output_name: '[Repolinter] Tier 1 OSS Policy Issues'
+
+          # The name to use for the issue label created by repolinter-action. This name
+          # should be unique to repolinter-action (i.e. not used by any other issue) to
+          # prevent repolinter-action from getting confused.
+          #
+          # This option will be ignored if output_type != "issue".
+          #
+          # Default: "repolinter"
+          label_name: 'Tier 1 Remediation'
```
Comment (decause-gov) [inline tier1/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:60]: let's make this label simpler, such as: `cms-oss-tier1`

File: tier2/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          output_name: '[Repolinter] Tier 2 OSS Policy Issues'
+
+          # The name to use for the issue label created by repolinter-action. This name
+          # should be unique to repolinter-action (i.e. not used by any other issue) to
+          # prevent repolinter-action from getting confused.
+          #
+          # This option will be ignored if output_type != "issue".
+          #
+          # Default: "repolinter"
+          label_name: 'Tier 2 Remediation'
```
Comment (decause-gov) [inline tier2/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:60]: same comment as above

File: tier3/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          output_name: '[Repolinter] Tier 3 OSS Policy Issues'
+
+          # The name to use for the issue label created by repolinter-action. This name
+          # should be unique to repolinter-action (i.e. not used by any other issue) to
+          # prevent repolinter-action from getting confused.
+          #
+          # This option will be ignored if output_type != "issue".
+          #
+          # Default: "repolinter"
+          label_name: 'Tier 3 Remediation'
```
Comment (decause-gov) [inline tier3/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:60]: same as line 60 above (cms-oss-tier3)

File: tier4/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          output_name: '[Repolinter] Tier 4 OSS Policy Issues'
+
+          # The name to use for the issue label created by repolinter-action. This name
+          # should be unique to repolinter-action (i.e. not used by any other issue) to
+          # prevent repolinter-action from getting confused.
+          #
+          # This option will be ignored if output_type != "issue".
+          #
+          # Default: "repolinter"
+          label_name: 'Tier 4 Remediation'
```
Comment (decause-gov) [inline tier4/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:60]: cms-oss-tier4

File: tier1/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+          output_type: 'issue'
+
+          # The title to use for the issue created by repolinter-action. This title
+          # should indicate the purpose of the issue, as well as that it was created by
+          # a bot.
+          #
+          # This option will be ignored if output_type != "issue".
+          #
+          # Default: "[Repolinter] Open Source Policy Issues"
+          output_name: '[Repolinter] Tier 1 OSS Policy Issues'
```
Comment (decause-gov) [inline tier1/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:51]: Let's try maybe: `[Repolinter] Tier 1 Repository Hygiene Issue`?

**Review by decause-gov — APPROVED** [review]
Some minor language tweaking around the ticket title and labels, but once we settle on those, excited to ship! 🚢

File: tier1/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+
+jobs:
+  resolve-repolinter-json:
+    uses: DSACMS/repo-scaffolder/.github/workflows/extendJSONFile.yml@add-repolinter-workflows
+    with: 
+      url_to_json: 'https://raw.githubusercontent.com/DSACMS/repo-scaffolder/main/tier1/%7B%7Bcookiecutter.project_slug%7D%7D/repolinter.json'
+  
+  repolinter-checks:
+    name: Tier 3 Checks
```
Comment (natalialuzuriaga) [inline tier1/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:14]: Rename to correct tier

File: tier2/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+
+jobs:
+  resolve-repolinter-json:
+    uses: DSACMS/repo-scaffolder/.github/workflows/extendJSONFile.yml@add-repolinter-workflows
+    with: 
+      url_to_json: 'https://raw.githubusercontent.com/DSACMS/repo-scaffolder/main/tier2/%7B%7Bcookiecutter.project_slug%7D%7D/repolinter.json'
+  
+  repolinter-checks:
+    name: Tier 3 Checks
```
Comment (natalialuzuriaga) [inline tier2/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:14]: Rename to correct tier

File: tier4/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+
+jobs:
+  resolve-repolinter-json:
+    uses: DSACMS/repo-scaffolder/.github/workflows/extendJSONFile.yml@add-repolinter-workflows
+    with: 
+      url_to_json: 'https://raw.githubusercontent.com/DSACMS/repo-scaffolder/main/tier4/%7B%7Bcookiecutter.project_slug%7D%7D/repolinter.json'
+  
+  repolinter-checks:
+    name: Tier 3 Checks
```
Comment (natalialuzuriaga) [inline tier4/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:14]: Rename to correct tier

File: tier1/{{cookiecutter.project_slug}}/.github/workflows/checks.yml [hunk 9]
```diff
@@ -0,0 +1,69 @@
+name: "run-linting-checks"
+on:
+  pull_request:
+    branches: [main, dev]
+
+
+jobs:
+  resolve-repolinter-json:
+    uses: DSACMS/repo-scaffolder/.github/workflows/extendJSONFile.yml@add-repolinter-workflows
```
Comment (natalialuzuriaga) [inline tier1/{{cookiecutter.project_slug}}/.github/workflows/checks.yml:9]: Looks like `extendJSONFile.yml` is on dev and not main. For this PR, could you merge into dev? I'll do a merge to main after

**Review by natalialuzuriaga — COMMENTED** [review]
Minor nitpicks but otherwise LGTM

**Pull request merged**